app/src/main/java/com/github/mobile/ui/search/SearchActivity.java

```diff
@@ -105,20 +123,26 @@ protected void onNewIntent(Intent intent) {
         handleIntent(intent);
     }
 
-    @Override
-    public boolean onSearchRequested() {
-        startSearch(query, true, null, false);
-        return true;
-    }
-
     private void handleIntent(Intent intent) {
-        if (ACTION_SEARCH.equals(intent.getAction()))
-            search(intent.getStringExtra(QUERY));
+        if (ACTION_SEARCH.equals(intent.getAction())) {
+            String query = intent.getStringExtra(QUERY);
+
+            // Update the SearchView's query
+            searchView.setQuery(query, false);
+
+            // Prevents the suggestions dropdown from showing after we submit
+            searchView.post(new Runnable() {
+                @Override public void run() {
+                    searchView.clearFocus();
+                }
+            });
+
+            search(query);
+        }
     }
 
     private void search(final String query) {
-        this.query = query;
-        getSupportActionBar().setTitle(query);
+        getIntent().putExtra(QUERY, query);
```

Comment:
Oops, that was a remnant of when I was approaching search differently. It's unneeded now.